src/main/scala/chisel3/aop/Select.scala

```diff
@@ -0,0 +1,423 @@
+// See LICENSE for license details.
+
+package chisel3.aop
+
+import chisel3._
+import chisel3.experimental.{BaseModule, FixedPoint}
+import chisel3.internal.HasId
+import chisel3.internal.firrtl._
+import firrtl.annotations.ReferenceTarget
+
+import scala.collection.mutable
+
+import scala.reflect.runtime.universe.TypeTag
+
+
+/** Use to select Chisel components in a module, after that module has been constructed
+  * Useful for adding additional Chisel annotations or for use within an [[Aspect]]
+  */
+object Select {
+
+  /** Return just leaf components of expanded node
+    *
+    * @param d Component to find leafs if aggregate typed. Intermediate fields/indicies are not included
+    * @return
+    */
+  def getLeafs(d: Data): Seq[Data] = d match {
+    case b: Bundle => b.getElements.flatMap(getLeafs)
+    case v: Vec[_] => v.getElements.flatMap(getLeafs)
+    case other => Seq(other)
+  }
+
+  /** Return all expanded components, including intermediate aggregate nodes
+    *
+    * @param d Component to find leafs if aggregate typed. Intermediate fields/indicies ARE included
+    * @return
+    */
+  def getIntermediateAndLeafs(d: Data): Seq[Data] = d match {
+    case b: Bundle => b +: b.getElements.flatMap(getIntermediateAndLeafs)
+    case v: Vec[_] => v +: v.getElements.flatMap(getIntermediateAndLeafs)
+    case other => Seq(other)
+  }
+
+
+  /** Collects all components selected by collector within module and all children modules it instantiates
+    *   directly or indirectly
+    * Accepts a collector function, rather than a collector partial function (see [[collectDeep]])
+    * @param module Module to collect components, as well as all children module it directly and indirectly instantiates
+    * @param collector Collector function to pick, given a module, which components to collect
+    * @param tag Required for generics to work, should ignore this
+    * @tparam T Type of the component that will be collected
+    * @return
+    */
+  def getDeep[T](module: BaseModule)(collector: BaseModule => Seq[T])(implicit tag: TypeTag[T]): Seq[T] = {
```

Comment:
Do you need the `TypeTag` here? Same thing for `getDeep` with the partial function parameter.

Reply:
Yeah you're right, its unnecessary.